subsys/bluetooth/services/ots/ots_oacp.c

```diff
@@ -298,102 +298,91 @@ static enum bt_gatt_ots_oacp_res_code oacp_proc_validate(
 	}
 };
 
-static enum bt_gatt_ots_oacp_res_code oacp_command_decode(
-	const uint8_t *buf, uint16_t len,
-	struct bt_gatt_ots_oacp_proc *proc)
+static int oacp_command_decode(const uint8_t *buf, uint16_t len,
+			       struct bt_gatt_ots_oacp_proc *proc)
 {
 	struct net_buf_simple net_buf;
 
 	net_buf_simple_init_with_data(&net_buf, (void *) buf, len);
 
+	__ASSERT(len >= OACP_PROC_TYPE_SIZE,
+		 "Cannot decode OACP procedure of size %u which is shorter than %u.",
+		 len, OACP_PROC_TYPE_SIZE);
+
 	proc->type = net_buf_simple_pull_u8(&net_buf);
 	switch (proc->type) {
 #if defined(CONFIG_BT_OTS_OACP_CREATE_SUPPORT)
 	case BT_GATT_OTS_OACP_PROC_CREATE:
+		if (net_buf.len < BT_GATT_OTS_OACP_CREATE_GENERIC_PARAMS_SIZE) {
+			return -EBADMSG;
+		}
 		proc->create_params.size = net_buf_simple_pull_le32(&net_buf);
-		bt_uuid_create(&proc->create_params.type.uuid, net_buf.data,
-			       net_buf.len);
+		if (!bt_uuid_create(&proc->create_params.type.uuid, net_buf.data,
+				    net_buf.len)) {
+			return -EBADMSG;
+		}
 		net_buf_simple_pull_mem(&net_buf, net_buf.len);
-		return BT_GATT_OTS_OACP_RES_SUCCESS;
+		if (proc->create_params.type.uuid.type == BT_UUID_TYPE_32) {
+			return -ENOTSUP;
+		}
+
+		return 0;
```

Comment:
We are not missing length checks based on whether `proc->create_params.type.uuid.type` is `BT_UUID_SIZE_16` or `BT_UUID_SIZE_128`?

i.e. if type is `BT_UUID_SIZE_16`, then `len` should be `BT_GATT_OTS_OACP_CREATE_UUID16_PARAMS_SIZE` (or `net_buf.len should be `BT_UUID_SIZE_16`).

Reply:
I did consider that. The `bt_uuid_create` attempts to create a UUID from the entire `net_buf` since it is told to use `net_buf.len`. If the function succeeds, it is because it created a 16-bit, 32-bit, or 128-bit UUID. If there wasn't enough or too many bytes to create any of these it would fail. Then we explicitly check that the 32-bit variant is not constructed.

Reply:
I think I am missing something. The implementation of `bt_uuid_create` I am looking at is:
```c
bool bt_uuid_create(struct bt_uuid *uuid, const uint8_t *data, uint8_t data_len)
{
	/* Copy UUID from packet data/internal variable to internal bt_uuid */
	switch (data_len) {
	case BT_UUID_SIZE_16:
		uuid->type = BT_UUID_TYPE_16;
		BT_UUID_16(uuid)->val = sys_get_le16(data);
		break;
	case BT_UUID_SIZE_32:
		uuid->type = BT_UUID_TYPE_32;
		BT_UUID_32(uuid)->val = sys_get_le32(data);
		break;
	case BT_UUID_SIZE_128:
		uuid->type = BT_UUID_TYPE_128;
		memcpy(&BT_UUID_128(uuid)->val, data, 16);
		break;
	default:
		return false;
	}
	return true;
}
```
`data_len` is the remaining length in the buffer that is yet to be parsed. If the length doesn't exactly match the 16-bit, 32-bit, or 128-bit UUID sizes, the function will not access the buffer at all.

Reply:
No, I'm sorry, I was the one that missed something :) Thanks